pkg/terminal/command.go

```diff
@@ -917,11 +918,58 @@ func (t *Term) formatGoroutine(g *api.Goroutine, fgl formatGoroutineLoc) string
 		locname = "Start"
 		loc = g.StartLoc
 	}
-	thread := ""
+
+	buf := new(bytes.Buffer)
+	fmt.Fprintf(buf, "%d - %s: %s", g.ID, locname, t.formatLocation(loc))
 	if g.ThreadID != 0 {
-		thread = fmt.Sprintf(" (thread %d)", g.ThreadID)
+		fmt.Fprintf(buf, " (thread %d)", g.ThreadID)
 	}
-	return fmt.Sprintf("%d - %s: %s%s", g.ID, locname, t.formatLocation(loc), thread)
+
+	if (g.Status == api.GoroutineWaiting || g.Status == api.GoroutineSyscall) && g.WaitReason != 0 {
+		var wr string
+		if g.WaitReason > 0 && g.WaitReason < int64(len(waitReasonStrings)) {
+			wr = waitReasonStrings[g.WaitReason]
+		} else {
+			wr = fmt.Sprintf("unknown wait reason %d", g.WaitReason)
+		}
+		fmt.Fprintf(buf, " [%s", wr)
+		if g.WaitSince > 0 {
+			fmt.Fprintf(buf, " %s", time.Since(time.Unix(0, g.WaitSince)).String())
+		}
+		fmt.Fprintf(buf, "]")
+	}
+
+	return buf.String()
+}
+
+var waitReasonStrings = [...]string{
```

Comment:
Is there ever a case we'd want this string outside of just the terminal? For example, returning wait reason as a string via API calls?

Reply:
As a potential consumer of the API (and the one that added support initially for this), I second @aarzilli's message.

Reply:
Ack, makes sense.